crypto/conf/conf_mod.c

```diff
@@ -642,6 +662,24 @@ char *CONF_get1_default_config_file(void)
         return NULL;
     BIO_snprintf(file, size, "%s%s%s", t, sep, OPENSSL_CONF);
 
+    /* if file not found, create path based on shared library with our symbol */
+    if (stat(file, &s) == -1) {
+        if (dladdr(OpenSSL_version, &info)) {
```

Comment:
How will this work on windows, or other platforms that don't have the dladdr call?

Additionally,, is info.dll_fname set when linked statically, I don't think the output will be what you expect

```
#define _GNU_SOURCE
#include <dlfcn.h>
#include <stdlib.h>
#include <stdio.h>
#include <openssl/crypto.h>


int main (int argc, char **argv)
{
    int rc;
    Dl_info info;

    rc = dladdr(OpenSSL_version, &info);
    if (rc != 0) {
        fprintf(stdout, "OpenSSL_version is located at %s\n", info.dli_fname);
    }
    exit(0);
}
```

```
./a.out
OpenSSL_version is located at ./a.out

cd ..
./openssl/a.out
OpenSSL_version is located at ./openssl/a.out

cd openssl/test
../a.out
OpenSSL_version is located at ../a.out
```


So any statically linked binary that doesn't set OPENSSL_CONF is going to look for the config file using whatever path you supplied to it when running the binary.  That doesn't seem like what you want